### ISSUE TITLE
S124: Comment Regular Expression Case Insensitive

### ISSUE THREAD

**Issue by iadcode** [SonarSource/SonarJS#1935]
With SonarJS 6.1, we have a custom rule from template S124 to track comment with a regular expression of ``(?i).*huh.*``. The ``(?!)`` bit was for case insensitive matching and was following what's documented in the rule description in SonarJS 6.1, and https://rules.sonarsource.com/javascript/RSPEC-124. 

> Note that, in order to match REVIEW regardless of the case, the ``(?i)`` modifier should be prepended to the expression, as in ``(?i).*REVIEW.*``.

It worked fine with SonarJS 6.1.
 
With SonarJS 6.2, scanning with this rule raises an error 
```
[error] SyntaxError: Error while loading rule 'comment-regex': Invalid regular expression: /(?i).*huh.*/: Invalid group
```
The documentation about case insensitive matching has also been removed from the rule description of S124.

Can you please advise if removing support for ``(?!)`` is an intentional change? If so, it might be clearer to update documentation in https://rules.sonarsource.com/javascript/RSPEC-124 and S124 to provide alternative examples.

**Comment by vilchik-elena** [comment]
Hi,

Thanks for the feedback, unfortunately indeed we dropped support for case-insensitivity without an alternative. We are currently discussing how better bring it back.

In the meantime you can change your regex to `.*[hH][uU][hH].*`, not great but should be ok for such small regex as yours.

sorry for the inconvenience!

**Comment by iadcode** [comment]
Thanks @vilchik-elena , what you suggested was indeed what we are using for now. One related point is that SonarJava's equivalent also uses the `(?!)` modifier for case insensitive matching.

**Comment by vilchik-elena** [comment]
The problem is that we are now relying on js-based regex evaluation, unlike java or js in the previous versions.

**Comment by p3k** [comment]
fwiw case-insensitive js regexes can be defined these ways:

```js
let regex1 = new RegExp('foobar', 'i');
let regex2 = /foobar/i;
```
